vehicle/polestar/identity.go

```diff
@@ -137,3 +149,35 @@ func (v *Identity) RefreshToken(token *oauth2.Token) (*oauth2.Token, error) {
 
 	return v.login()
 }
+
+func (v *Identity) confirmConsentAndGetCode(resume string, uid string) (string, error) {
+	// Extract the user ID (UID) from the redirect parameters
+	if uid == "" {
+		return "", fmt.Errorf("Failed to extract user ID: %s", uid)
+	}
+
+	// Confirm user consent by submitting the consent form, which rejects cookies
+	data := url.Values{
+		"pf.submit": []string{"true"},
+		"subject":   []string{uid},
+	}
+
+	// Retrieve the authorization code after consent has been confirmed
+	var param request.InterceptResult
+	v.Client.CheckRedirect, param = request.InterceptRedirect("code", true)
+	defer func() { v.Client.CheckRedirect = nil }()
+
+	// Make a POST request to confirm the user consent
+	if _, err := v.Post(fmt.Sprintf("%s/as/%s/resume/as/authorization.ping", OAuthURI, resume), request.FormContent, strings.NewReader(data.Encode())); err != nil {
+		return "", fmt.Errorf("Error confirming user consent to reject cookies during the authentication process: %w", err)
+	}
+
+	// Extract the authorization code from the response
+	code, err := param()
+	if err != nil || code == "" {
+		return "", fmt.Errorf("Failed to extract authorization code: %w", err)
```

Comment:
```suggestion
		return "", fmt.Errorf("failed extracting authorisation code: %w", err)
```

Reply:
Not sure %w wrapping a nil error will not panic.

Reply:
No, it does not panic